src/SFA.DAS.ApprenticeCommitments/Application/Commands/ApprenticeshipCreated/ApprenticeshipCreatedCommandHandler.cs

```diff
@@ -36,10 +33,52 @@ public class CreateApprenticeshipCommandHandler : IRequestHandler<CreateApprenti
         }
 
         public async Task<Unit> Handle(
-            CreateApprenticeshipCommand command,
+            ApprenticeshipCreatedCommand command,
             CancellationToken cancellationToken)
         {
-            var (trainingProvider, apprentice, course) = await GetExternalData(command);
+            var (trainingProvider, apprenticeship, course) = await GetExternalData(command);
+
+            if (apprenticeship.ContinuationOfId == null)
+            {
+                await CreateApprenticeshipAndSendInvitation(command, trainingProvider, course, apprenticeship);
+            }
+            else
+            {
+                await ChangeApprenticeship(command, trainingProvider, course, apprenticeship);
+            }
+            return Unit.Value;
+        }
+
+        private async Task ChangeApprenticeship(ApprenticeshipCreatedCommand command, TrainingProviderResponse trainingProvider,
```

Comment:
Is there much difference between this `ChangeApprenticeship` and the one that happens because of an `ApprenticeshipUpdatedApprovedEvent`?

I suspect there may be a way to make that clearer in the code. I was already thinking of removing some duplication between the Create and Update handlers - they both do essentially the same thing.

Reply:
No there isn't much difference, but we only know its a continuation apprenticeship in the handler

Reply:
I'm happy to refactor the GetExternalData function into a Extension method